src/bacnet/bacstr.c

```diff
@@ -586,6 +586,29 @@ bool characterstring_ansi_same(BACNET_CHARACTER_STRING *dest, const char *src)
     return same_status;
 }
 
+/**
+ * Returns number of UTF8 code points in a character string.
+ *
+ * @param dest  Pointer to the string to count the UTF8 code points.
+ *
+ * @return Length of the character string in utf8 codepoints
+ */
+size_t characterstring_utf8_length(const BACNET_CHARACTER_STRING *str)
+{
+    size_t count = 0;
+    int i = 0;
+
+    while ((i < MAX_CHARACTER_STRING_BYTES) && (str->value[i] != '\0')) {
+        if ((str->value[i] & 0xc0) != 0x80) {
```

Comment:
Just a question, what is the significance of anding the value with 0xc0, and also output of that not being equal to 0x80?

Reply:
I'm guessing this is checking 2 flags, one should be set, one should be clear. a comment on this or use of a macro may be useful here.